pkg/apis/certmanager/v1alpha1/types_issuer.go

```diff
@@ -504,8 +504,14 @@ type ACMEIssuerDNS01ProviderRoute53 struct {
 // ACMEIssuerDNS01ProviderAzureDNS is a structure containing the
 // configuration for Azure DNS
 type ACMEIssuerDNS01ProviderAzureDNS struct {
+	// The ClientID of the service principal used to authenticate with the Azure API
+	// Not required if using Managed Identities
+	// +optional
 	ClientID string `json:"clientID"`
 
+	// The secret key for the service principal.
+	// Required if ClientID is configured
+	// +optional
 	ClientSecret SecretKeySelector `json:"clientSecretSecretRef"`
 
 	SubscriptionID string `json:"subscriptionID"`
```

Comment:
Just confirming, this field is still required even when using managed identity? Also, if you are familiar with these options, it'd be awesome if you could add a comment to any fields missing one to help others in future 😄

Reply:
sure, i'll add comments explaining them :+1:

`ClientSecret` is not used when with managed identity